lib/IAMClient.js

```diff
@@ -395,13 +400,33 @@ class IAMClient {
                 ret += receivedData.toString();
             })
 
-            .on('error', callback)
+            .on('error', err => {
+                if (logger) {
+                    logger.debug('error receiving data', {
```

Comment:
i put it to debug to let the main component taking care of the internal error logging :) and yes the check is necessary since the logger is not always defined, its not used for other routes than s3 for example